### PR TITLE
[docs] Update m3db operator docs with v0.13.0 features

### PR DESCRIPTION
**Pull request by EdgeJ**
**What this PR does / why we need it**:
Updates the documentation for the m3db-operator to include API params for the latest version (v0.13.0) and includes examples for using the pod anti-affinity settings.  Also fixes a few broken internal links on the page.

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

NONE

**Does this PR require updating code package or user-facing documentation?**:

NONE

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
Thanks @EdgeJ It's concerning that those links got through our link checker :(

File: site/content/operator/configuration/node_affinity.md [hunk 1]
```diff
@@ -98,6 +98,45 @@ spec:
     effect: NoSchedule
     operator: Exists
 ```
+## Pod Anti-Affinity
+
+Kubernetes also provides another way to repel pods by using pod
```
Comment (ChrisChinchilla) [inline site/content/operator/configuration/node_affinity.md:103]:
```suggestion
Kubernetes provides another way to repel pods by using pod
```

File: site/content/operator/configuration/node_affinity.md [hunk 2]
```diff
@@ -98,6 +98,45 @@ spec:
...
+[anti-affinity][k8s-anti-affinity] to prevent multiple pods from scheduling on
+the same node. IsolationGroups can be configured in such a way that the pods
+created in the group will not be scheduled on a Kubernetes node with each
+other, potentially causing resource contention and reducing high durability.
+```yaml
```
Comment (ChrisChinchilla) [inline site/content/operator/configuration/node_affinity.md:108]:
```suggestion

```yaml
```

**Comment by ChrisChinchilla** [comment]
@EdgeJ Couple of small changes requested

**Comment by EdgeJ** [comment]
> Thanks @EdgeJ It's concerning that those links got through our link checker :(

I believe they probably sneakedy by because the links still "worked", they just linked to the wrong place.

**Comment by EdgeJ** [comment]
Updated PR with requested changes and rebased against upstream/master.

**Review by ChrisChinchilla — APPROVED** [review]
Thanks for the contribution @EdgeJ

**Pull request merged**